pkg/virt-controller/watch/preset.go

```diff
@@ -143,10 +144,16 @@ func (c *VirtualMachinePresetController) initializeVirtualMachine(vm *kubev1.Vir
 	matchingPresets := filterPresets(allPresets, vm, c.recorder)
 
 	if len(matchingPresets) != 0 {
-		applyPresets(vm, matchingPresets, c.recorder)
+		success = applyPresets(vm, matchingPresets, c.recorder)
 	}
 
-	logger.Object(vm).Info("Marking VM as initialized and updating")
+	if !success {
+		logger.Object(vm).Warning("Marking VM as failed")
+		vm.Status.Phase = kubev1.Failed
```

Comment:
Don't set status to fail if no matchingPresets are present. This is causing functional tests to fail on my machine.

Reply:
Good catch! The issue was `success` was never initialized so it defaulted to `false`. Unit and functional test cases added.